panel/src/components/Forms/Toolbar/LinkDialog.vue

```diff
@@ -65,6 +61,28 @@ export default {
 				return `<${this.value.url}>`;
 			}
 		},
+		open(input, selection) {
+			this.parseDOM(selection);
+			this.$refs.dialog.open();
+		},
+		parseDOM(selection) {
+			if (selection?.length === 0) {
+				this.value.text = null;
+				this.value.url  = null;
+				return;
+			}
+
+			const regexp = new RegExp(`^\\(link:\\s+?(.*?)\\s+?(?:text:\\s+?(.*?))?\\)$`, "gis");
```

Comment:
I think this can be replaced with this to avoid some escaping:

```suggestion
			const regexp = /^\(link:\s+?(.*?)\s+?(?:text:\s+?(.*?))?\)$/gis;
```

However I'd make the following changes:

```suggestion
			const regexp = /^\(link:\s*(.*?)(?:\s*text:\s*(.*?))?\)$/is;
```

Because:

- the spaces after tag names are not required
- greedy matches for the spaces mean that multiple spaces are trimmed away instead of being matched by the following capture group
- global flag is not needed as we match the whole string